### PR TITLE
rt: several minor updates to aspec and ainvs

### PR DESCRIPTION
**Pull request by corlewis**
The most significant change is updating `fast_finalise` to not handle the `SchedContextCap` case.

### CODE REVIEW

**Review by MitchellBuckley — APPROVED** [review]
Looks good!

File: proof/invariant-abstract/KHeap_AI.thy [hunk 1]
```diff
@@ -87,7 +87,7 @@ method get_simple_ko_method =
   (wpsimp simp: get_simple_ko_def partial_inv_def the_equality split: kernel_object.splits)
 
 lemma get_simple_ko_wp:
-  "\<lbrace>\<lambda>s. \<forall>ntfn. ko_at (f ntfn) ntfnptr s \<longrightarrow> P ntfn s\<rbrace> get_simple_ko f ntfnptr \<lbrace>P\<rbrace>"
+  "\<lbrace>\<lambda>s. \<forall>obj. ko_at (f obj) ptr s \<longrightarrow> P obj s\<rbrace> get_simple_ko f ptr \<lbrace>P\<rbrace>"
```
Comment (mktnk3) [inline proof/invariant-abstract/KHeap_AI.thy:90]: Nice cleanup!

**Review by mktnk3 — APPROVED** [review]
thanks for catching the redundant SchedContext case in fast_finalise :)

**Review by michaelmcinerney — APPROVED** [review]

**Review by lsf37 — APPROVED** [review]
Nice!

**Pull request merged**